### PR TITLE
[quick_actions] Support v2 android embedder.

### PR DESCRIPTION
**Pull request by cyanglaz**
## Description

Minimum work to migrate to the android v2 embedding.

## Related Issues

https://github.com/flutter/flutter/issues/41845

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/quick_actions/android/src/main/java/io/flutter/plugins/quickactions/QuickActionsPlugin.java [hunk 1]
```diff
@@ -4,130 +4,70 @@
...
 
   /**
    * Plugin registration.
    *
    * <p>Must be called when the application is created.
    */
   public static void registerWith(Registrar registrar) {
-    final MethodChannel channel = new MethodChannel(registrar.messenger(), CHANNEL_ID);
-    channel.setMethodCallHandler(new QuickActionsPlugin(registrar));
+    QuickActionsPlugin plugin = new QuickActionsPlugin();
```
Comment (bparrishMines) [inline packages/quick_actions/android/src/main/java/io/flutter/plugins/quickactions/QuickActionsPlugin.java]: nit:
```suggestion
    final QuickActionsPlugin plugin = new QuickActionsPlugin();
```

File: packages/quick_actions/android/src/main/java/io/flutter/plugins/quickactions/QuickActionsPlugin.java [hunk 2]
```diff
@@ -4,130 +4,70 @@
...
   private static final String CHANNEL_ID = "plugins.flutter.io/quick_actions";
-  private static final String EXTRA_ACTION = "some unique action key";
 
-  private final Registrar registrar;
-
-  private QuickActionsPlugin(Registrar registrar) {
-    this.registrar = registrar;
-  }
+  MethodChannel channel;
+  MethodCallHandlerImpl handler;
```
Comment (bparrishMines) [inline packages/quick_actions/android/src/main/java/io/flutter/plugins/quickactions/QuickActionsPlugin.java]:
```suggestion
  private MethodChannel channel;
  private MethodCallHandlerImpl handler;
```

File: packages/quick_actions/android/src/main/java/io/flutter/plugins/quickactions/MethodCallHandlerImpl.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+import io.flutter.plugin.common.MethodCall;
+import io.flutter.plugin.common.MethodChannel;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+class MethodCallHandlerImpl implements MethodChannel.MethodCallHandler {
+
+  private static final String CHANNEL_ID = "plugins.flutter.io/quick_actions";
+  private static final String EXTRA_ACTION = "some unique action key";
```
Comment (matthew-carroll) [inline packages/quick_actions/android/src/main/java/io/flutter/plugins/quickactions/MethodCallHandlerImpl.java]: Should "some unique action key" be written here?

Comment (cyanglaz) [reply packages/quick_actions/android/src/main/java/io/flutter/plugins/quickactions/MethodCallHandlerImpl.java]: I don't have enough context to give input here. It seems this is introduced in https://github.com/flutter/plugins/commit/63609c94779331977d9492fc89cd6fb56cc68256 
and @collinjackson or @BugsBunnyBR might be able to give some inputs on this.
Anyways, this is not related to this migration. I suggest we create new issue and fix it in a later PR if needs to be fixed.

Comment (BugsBunnyBR) [reply packages/quick_actions/android/src/main/java/io/flutter/plugins/quickactions/MethodCallHandlerImpl.java]: That is just a unique string, the value itself does not matter.

Comment (cyanglaz) [reply packages/quick_actions/android/src/main/java/io/flutter/plugins/quickactions/MethodCallHandlerImpl.java]: @BugsBunnyBR We can make the value of the String unambiguous so it is more readable.

**Pull request merged**